crates/rust-analyzer/src/config.rs

```diff
@@ -2965,7 +3021,7 @@ mod tests {
     #[test]
     fn generate_config_documentation() {
         let docs_path = project_root().join("docs/user/generated_config.adoc");
-        let expected = ConfigData::manual();
+        let expected = ConfigInput::manual();
```

Comment:
small nit: there's been a pattern of using `Data` inside of rust-analyzer for things that implement `serde::Deserialize`, and while I don't see the larger picture that this branch is painting, I think it might better to keep this suffixed with `Data` instead of `Input`.

Reply:
There already is a ConfigData. What should this one be called, given it contains an Option version of every field in ConfigData?

Reply:
Ah, I'm sorry. I assumed that `ConfigInput` _was_ `ConfigData`; I didn't realize it was a separate struct with `Options` for fields.

Disregard the nit then!